internal/provider/resource_project_domain.go

```diff
@@ -0,0 +1,199 @@
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"created_at": {
+				Description: "A number containing the date when the project domain was created in milliseconds.",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"updated_at": {
+				Description: "A number containing the date when the project domain was updated in milliseconds.",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceProjectDomainCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*vercel.Client)
+
+	projectDomain := project_domain.CreateProjectDomain{
+		Name: d.Get("name").(string),
+	}
+
+	if redirect, ok := d.GetOk("redirect"); ok {
+		projectDomain.Redirect = redirect.(string)
+	}
+
+	if redirectStatusCode, ok := d.GetOk("redirect_status_code"); ok {
+		projectDomain.RedirectStatusCode = redirectStatusCode.(int)
+	}
+
+	if gitBranch, ok := d.GetOk("git_branch"); ok {
+		projectDomain.GitBranch = gitBranch.(string)
+	}
+
+	id, err := client.ProjectDomain.Create(d.Get("project_id").(string), projectDomain, d.Get("team_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set("redirect", projectDomain.Redirect)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	err = d.Set("redirect_status_code", projectDomain.RedirectStatusCode)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(id)
+
+	return resourceProjectDomainRead(ctx, d, meta)
+}
+
+func resourceProjectDomainRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*vercel.Client)
+
+	id := d.Id()
+
+	subs := strings.SplitN(id, ":", 2)
+	projectID, name := subs[0], subs[1]
+
+	projectDomain, err := client.ProjectDomain.Read(projectID, name, d.Get("team_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set("name", projectDomain.Name)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	err = d.Set("project_id", projectDomain.ProjectID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	err = d.Set("created_at", projectDomain.CreatedAt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	err = d.Set("updated_at", projectDomain.UpdatedAt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+func resourceProjectDomainUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*vercel.Client)
+	var update project_domain.UpdateProjectDomain
+
+	id := d.Id()
+	subs := strings.SplitN(id, ":", 2)
+	projectID, name := subs[0], subs[1]
+
+	if d.HasChange("redirect") {
+		update.Redirect = d.Get("redirect").(string)
+	}
+
+	projectDomain, err := client.ProjectDomain.Update(projectID, name, update, d.Get("team_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set("name", projectDomain.Name)
```

Comment:
is there a reason these are all set individually instead of at once?

Reply:
So we can check for errors for each one. Pretty standard go pattern.